java/vector/src/test/java/org/apache/arrow/vector/ipc/TestArrowStream.java

```diff
@@ -99,4 +99,56 @@ public void testReadWrite() throws IOException {
       }
     }
   }
+
+  @Test
+  public void testReadWriteMultipleBatches() throws IOException {
+
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+    try (IntVector vector = new IntVector("foo", allocator);) {
+      Schema schema = new Schema(Collections.singletonList(vector.getField()), null);
+      try (VectorSchemaRoot root = new VectorSchemaRoot(schema, Collections.singletonList((FieldVector) vector), vector.getValueCount());
+           ArrowStreamWriter writer = new ArrowStreamWriter(root, new DictionaryProvider.MapDictionaryProvider(), Channels.newChannel(os));) {
+        writer.start();
+
+        vector.setNull(0);
+        vector.setSafe(1, 1);
+        vector.setSafe(2, 2);
+        vector.setNull(3);
+        vector.setSafe(4, 1);
+        vector.setValueCount(5);
+        root.setRowCount(5);
+        writer.writeBatch();
+
+        vector.setNull(0);
+        vector.setSafe(1, 1);
+        vector.setSafe(2, 2);
+        vector.setValueCount(3);
+        root.setRowCount(3);
+        writer.writeBatch();
+      }
+    }
+
+    ByteArrayInputStream in = new ByteArrayInputStream(os.toByteArray());
+
+    try (ArrowStreamReader reader = new ArrowStreamReader(in, allocator);) {
+      IntVector read = (IntVector) reader.getVectorSchemaRoot().getFieldVectors().get(0);
+
+      reader.loadNextBatch();
+
+      assertEquals(read.getValueCount(), 5);
+      assertNull(read.getObject(0));
+      assertEquals(read.getObject(1), Integer.valueOf(1));
```

Comment:
you can just use the `get` method that returns an integer

Reply:
these were from my original test - we use getObject as we don't care about the runtime type of the value, and we need to wrap the primitive anyway.